### PR TITLE
ENGG-2326: Empty state in env variable screen

### PR DESCRIPTION
**Pull request by RuntimeTerror10**

### CODE REVIEW

**Comment by rohanmathur91** [comment]
Is this behaviour expected?, if no env created then it should go to empty state?

https://github.com/user-attachments/assets/96b59963-74c9-4a5e-ab72-75cae8bea596

File: app/src/backend/environment/hooks/useEnvironmentManager.ts [hunk 1]
```diff
@@ -61,23 +61,13 @@ const useEnvironmentManager = (initListenerAndFetcher: boolean = false) => {
...
-              if (defaultEnv) {
-                setCurrentEnvironment(defaultEnv.id);
-              }
-            });
-          } else if (!currentEnvironmentId) {
-            // if there is no active environment, set the first environment as the active environment
-            const defaultEnvironment = Object.keys(environmentMap)[0];
-            setCurrentEnvironment(defaultEnvironment);
+          if (Object.keys(environmentMap).length > 0 && !environmentMap[currentEnvironmentId]) {
+            // setting the first environment as the current environment if the current environment is not found in environmentMap
```
Comment (rohanmathur91) [inline app/src/backend/environment/hooks/useEnvironmentManager.ts:65]: When will this case happen?

File: app/src/features/apiClient/screens/apiClient/components/sidebar/components/sidebarListHeader/sidebarListHeader.scss [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+  gap: 6px;
   padding: 8px 0;
 
   &-button {
     padding: 4px 8px;
   }
 
   &-search {
-    width: 200px;
+    width: -webkit-fill-available;
```
Comment (rohanmathur91) [inline app/src/features/apiClient/screens/apiClient/components/sidebar/components/sidebarListHeader/sidebarListHeader.scss:12]: Why no `100%`?

File: app/src/features/apiClient/screens/environment/components/VariablesList/VariablesList.tsx [hunk 3]
```diff
@@ -76,7 +76,8 @@ export const VariablesList: React.FC<VariablesListProps> = ({ searchValue, curre
             return acc;
           }, {});
 
-          setVariables(currentEnvironmentId, variablesToSave);
+          await setVariables(currentEnvironmentId, variablesToSave);
```
Comment (rohanmathur91) [inline app/src/features/apiClient/screens/environment/components/VariablesList/VariablesList.tsx:79]: Make sure the failed state is handled

File: app/src/features/apiClient/screens/environment/components/emptyEnvironmentView/EmptyEnvironmentView.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  const environments = getAllEnvironments();
+
+  const handleCreateNewEnvironment = () => {
+    if (!user.loggedIn) {
+      dispatch(
+        actions.toggleActiveModal({
+          modalName: "authModal",
+          newValue: true,
+          newProps: {
+            eventSource: "empty_environment_view",
```
Comment (rohanmathur91) [inline app/src/features/apiClient/screens/environment/components/emptyEnvironmentView/EmptyEnvironmentView.tsx:33]: Have a enum for all the event sources

File: app/src/features/apiClient/screens/environment/components/environmentView/EnvironmentView.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+  useEffect(() => {
+    if (!isEnvironmentsLoading) {
+      if (location.pathname.includes(PATHS.API_CLIENT.ENVIRONMENTS.NEW.RELATIVE)) {
+        return;
+      }
+      const environments = getAllEnvironments();
+      const hasAccessToEnvironment = environments?.some((env) => env.id === envId);
+      if (environments?.length === 0 || !user.loggedIn || !hasAccessToEnvironment) {
+        navigate(PATHS.API_CLIENT.ENVIRONMENTS.ABSOLUTE);
```
Comment (rohanmathur91) [inline app/src/features/apiClient/screens/environment/components/environmentView/EnvironmentView.tsx:29]: If user is not logged in we can navigate before getting all the environments

File: app/src/features/apiClient/screens/environment/components/environmentsList/EnvironmentsList.tsx [hunk 6]
```diff
@@ -52,6 +63,10 @@ export const EnvironmentsList = () => {
       setIsLoading(true);
       const newEnvironment = await addNewEnvironment(newEnvironmentValue);
       if (newEnvironment) {
+        if (environments.length === 0) {
+          // if there are no environments, set the new environment as the active environment
+          setCurrentEnvironment(newEnvironment.id);
+        }
```
Comment (rohanmathur91) [inline app/src/features/apiClient/screens/environment/components/environmentsList/EnvironmentsList.tsx:69]: We can handle this while saving the new environment (can be ignored)

**Review by rohanmathur91 — CHANGES_REQUESTED** [review]

File: app/src/features/apiClient/screens/apiClient/components/sidebar/components/sidebarListHeader/sidebarListHeader.scss (hunk 2 above)
Comment (RuntimeTerror10) [reply app/src/features/apiClient/screens/apiClient/components/sidebar/components/sidebarListHeader/sidebarListHeader.scss]: The list header can also have a 3 dot dropdown menu on the right hand side of the input. I've changed this to `flex:1` though.

File: app/src/backend/environment/hooks/useEnvironmentManager.ts (hunk 1 above)
Comment (RuntimeTerror10) [reply app/src/backend/environment/hooks/useEnvironmentManager.ts:65]: switching workspaces and auth state changes

**Comment by RuntimeTerror10** [comment]
> Is this behaviour expected?, if no env created then it should go to empty state?
> 
>  Screen.Recording.2024-10-24.at.7.08.47.PM.mov

Keeping it this way right now because we don't have the screen for creating a new env when user has already has some env present. Showing an empty table didn't look ideal to me

**Review by rohanmathur91 — APPROVED** [review]

**Pull request merged**